auth_saml/controllers/main.py

```diff
@@ -0,0 +1,289 @@
+# Copyright (C) 2020 GlodoUK <https://www.glodo.uk/>
+# Copyright (C) 2010-2016, 2022-2023 XCG Consulting <https://xcg-consulting.fr/>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import functools
+import json
+import logging
+
+import werkzeug.utils
+from werkzeug.exceptions import BadRequest
+from werkzeug.urls import url_quote_plus
+
+import odoo
+from odoo import SUPERUSER_ID, _, api, http, models, registry as registry_get
+from odoo.http import request
+
+from odoo.addons.web.controllers.home import Home, ensure_db
+from odoo.addons.web.controllers.utils import _get_login_redirect_url
+
+_logger = logging.getLogger(__name__)
+
+
+# ----------------------------------------------------------
+# helpers
+# ----------------------------------------------------------
+
+
+def fragment_to_query_string(func):
+    @functools.wraps(func)
+    def wrapper(self, **kw):
+        if not kw:
+            return """<html><head><script>
+                var l = window.location;
+                var q = l.hash.substring(1);
+                var r = '/' + l.search;
+                if(q.length !== 0) {
+                    var s = l.search ? (l.search === '?' ? '' : '&') : '?';
+                    r = l.pathname + l.search + s + q;
+                }
+                window.location = r;
+            </script></head><body></body></html>"""
+        return func(self, **kw)
+
+    return wrapper
+
+
+# ----------------------------------------------------------
+# Controller
+# ----------------------------------------------------------
+
+
+class SAMLLogin(Home):
+    def _list_saml_providers_domain(self):  # pylint: disable=no-self-use
```

Comment:
Whenever a lint is disabled, a small comment should be provided to explaing why is being is disabled and why it's not applicable for that specific case

Reply:
Added a comment and added it to 483 too.